design/queries.go

```diff
@@ -0,0 +1,104 @@
+package design
+
+import (
+	d "github.com/goadesign/goa/design"
+	a "github.com/goadesign/goa/design/apidsl"
+)
+
+var query = a.Type("Query", func() {
+	a.Description(`JSONAPI store for the data of a Query. See also http://jsonapi.org/format/#document-resource-object`)
+	a.Attribute("type", d.String, func() {
+		a.Enum("queries")
+	})
+	a.Attribute("id", d.UUID, "ID of query", func() {
+		a.Example("40bbdd3d-8b5d-4fd6-ac90-7236b669af04")
+	})
+	a.Attribute("attributes", queryAttributes)
+	a.Attribute("links", genericLinks)
+	a.Attribute("relationships", queryRelationships)
+	a.Required("type", "attributes")
+})
+
+var queryRelationships = a.Type("QueryRelations", func() {
+	a.Attribute("creator", relationGeneric, "This defines the creator of the query")
+})
+
+var queryAttributes = a.Type("QueryAttributes", func() {
+	a.Description(`JSONAPI store for all the "attributes" of a Query. See also http://jsonapi.org/format/#document-resource-object-attributes`)
+	a.Attribute("title", d.String, "The title of saved query", nameValidationFunction)
+	a.Attribute("created-at", d.DateTime, "When the query was created", func() {
+		a.Example("2016-11-29T23:18:14Z")
+	})
+	a.Attribute("updated-at", d.DateTime, "When the query was updated", func() {
```

Comment:
We don't plan to support query update, right? If so, why do we need this `updated-at` attribute?

Reply:
yes right, removed in https://github.com/fabric8-services/fabric8-wit/pull/1824/commits/fc533a888888f670fba44bd208905e629ba34cda